### DISCUSSION TITLE
Why is a monad a monoid in the category of endofunctors and why is that a problem?

### DISCUSSION THREAD

**Discussion by JustBobinAround** [anahi-hub/list-c#24] [Q&A]
Why is a monad a monoid in the category of endofunctors and why is that a problem?

**Comment by AmadNaseem** [comment]
A monad is like a way to combine actions using rules, but this can make understanding how steps and effects work in code more difficult.

**Comment by RH2004** [comment]
A monad is a monoid in the category of endofunctors because: Endofunctors can be composed

Function composition behaves like multiplication

There is an identity functor

A monad gives you:

- a way to combine functors (join / bind)

- an identity (return)
